perceiver/data/audio/symbolic.py

```diff
@@ -0,0 +1,216 @@
+import functools
+import multiprocessing
+import os
+import random
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+from perceiver.data.audio.midi_processor import encode_midi_files
+
+
+class SymbolicAudioDataModule(pl.LightningDataModule):
+    _EXAMPLE_SEPARATOR_INPUT_ID = -1
+    _PAD_INPUT_ID = 388
+    _VOCAB_SIZE = 389
+
+    def __init__(
+        self,
+        dataset_dir: str,
+        max_seq_len: int,
+        min_seq_len: Optional[int] = None,
+        padding_side: Optional[str] = "left",
+        batch_size: int = 16,
+        num_workers: int = 1,
+        preproc_workers: int = multiprocessing.cpu_count(),
```

Comment:
Make optional and initialize to `num_workers` to be consistent with other data modules.

Reply:
`preproc_workers` should be set to a high value as it controls the number of workers encoding the midi files, whereas there is no benefit of setting `num_workers > 1` as each dataset split is read from a single file.
How about we remove the `num_workers` option altogether?